app/services/force/access_token.rb

```diff
@@ -0,0 +1,95 @@
+require 'net/http'
+require 'uri'
+
+module Force
+  # Utility for requesting access tokens for login into saleforce
+  module AccessToken
+    # Token response
+    class Response
+      attr_reader :body
+      def initialize(body)
+        @body = JSON.parse(body)
+      end
+
+      def invalid?
+        @body.key?('error')
+      end
+
+      def token
+        @body['access_token']
+      end
+    end
+
+    class CannotGetTokenException < StandardError; end
+
+    # This should be use to get a new token, regardles of the underline method.
+    def self.request_new_token
+      request_new_with_restforce
+    end
+
+    ##########################################
+    # Request new token unsing Restforce lib
+    ##########################################
+
+    def self.request_new_with_restforce
+      client = restforce_client
+      client.authenticate!
+      client.options[:oauth_token]
+    end
+
+    def self.request_new_with_restforce_attributes
+      {
+        instance_url: EnvUtils.get!('SALESFORCE_INSTANCE_URL'),
+        username: EnvUtils.get!('SALESFORCE_USERNAME'),
+        password: EnvUtils.get!('SALESFORCE_PASSWORD'),
+        security_token: EnvUtils.get!('SALESFORCE_SECURITY_TOKEN'),
+        client_id: EnvUtils.get!('SALESFORCE_CLIENT_ID'),
+        client_secret: EnvUtils.get!('SALESFORCE_CLIENT_SECRET'),
+        api_version: EnvUtils.get!('SALESFORCE_API_VERSION'),
+        mashify: false,
+        authentication_retries: 1,
+        host: 'test.salesforce.com',
+      }
+    end
+
+    def self.restforce_client
+      Restforce.new(request_new_with_restforce_attributes)
+    end
+
+    ####################################################
+    # Request new token unsing HTTP post with oauth2
+    ####################################################
+
+    def self.request_new_with_http
```

Comment:
This is the old way we were getting the token. I thought about leaving the code as a reference in case we might need it in the future. We can remove it, it's not being used.

Reply:
Gotcha. Let's remove unused code to keep maintainability.